### PR TITLE
Resumable Upload: Requesting digests from server using `Want-Repr-Digest`

### PR DESCRIPTION
**Pull request by Acconut**
Closes https://github.com/httpwg/http-extensions/issues/2834.

A client can express its interest in digests using `Want-Repr-Digest` during upload creation. The server can then respond with `Repr-Digest` after POST and PATCH requests to share its digest with the client. The client computes its digest as well during the upload and compares the digest to spot issues. The representation digest of an upload resource covers the entire bytes that have been uploaded so far (i.e. right until the upload offset).

@ioggstream @LPardue Is this in line with the semantics of RFC 9530?

### CODE REVIEW

**Review by guoye-zhang — APPROVED** [review]

**Comment by ioggstream** [comment]
@Acconut Need a week for the review :)

File: draft-ietf-httpbis-resumable-upload.md [hunk 1]
```diff
@@ -510,10 +511,12 @@ Unlike `Content-Encoding` (see {{Section 8.4.1 of HTTP}}), `Transfer-Encoding` (
...
 
+In the other case, that the client does not know the integrity digest of the entire data upfront, it can ask the server to compute the integrity digests during the upload and send them back to the client. The client expresses its interest using the `Want-Repr-Digest` header field with its preferred algorithms when creating an upload ({{upload-creation}}). The server SHOULD compute the representation digests using the preferred algorithms once the upload is complete and include the corresponding `Repr-Digest` header field in the response. Alternatively, the server MAY compute the digest continuously during the upload and include the `Repr-Digest` header field in responses to upload creation ({{upload-creation}}) and upload appending requests ({{upload-appending}}) even when the upload is not completed yet. This allows the client to simultaneously compute the digest of the transmitted upload data, compare its digest to the server's digest, and spot data integrity issues. If an upload is spread across multiple requests, data integrity issues can be found even before the upload is fully completed.
```
Comment (ioggstream) [inline draft-ietf-httpbis-resumable-upload.md:518]:
```suggestion
When creating an upload (see {{upload-creation}}),
the client MAY ask the server to compute and return the integrity digests using a `Want-Repr-Digest` field conveying the preferred algorithms.
The response SHOULD include at least one of the requested digests,
but MAY not include it. <!-- If the server is overloaded -->

The server SHOULD compute the representation digests using the preferred algorithms once the upload is complete and include the corresponding `Repr-Digest` header field in the response. Alternatively, the server MAY compute the digest continuously during the upload and include the `Repr-Digest` header field in responses to upload creation ({{upload-creation}}) and upload appending requests ({{upload-appending}}) even when the upload is not completed yet. This allows the client to simultaneously compute the digest of the transmitted upload data, compare its digest to the server's digest, and spot data integrity issues. If an upload is spread across multiple requests, data integrity issues can be found even before the upload is fully completed.
```

**Comment by ioggstream** [comment]
Preliminary feedback:

1. I'd simplify the language/split text on multiple lines;
2. Describe processing in separate subsections, e.g.

```
## Reply on finish

...


## Reply in transit

...

```

**Comment by Acconut** [comment]
Thank you for the feedback, Roberto. I added subsections for representation and content digests in https://github.com/httpwg/http-extensions/pull/2917/commits/664679150c9ece5f5f5cb3b89b3b398fb225841b. I think this division helps the reader and also makes sense since representation digests can cover the entire upload, while content digests only cover one request. Let me know if you think differently.

**Pull request merged**